socialapp/src/components/Profile/FollowButton.jsx

```diff
@@ -36,31 +66,56 @@ export default function FollowButton(props) {
 
   useEffect(() => {
     const following = async () => {
-      await api
-        .get(`${baseURL}/authors/${author_id}/followers/${currentAuthor.uuid}`)
-        .then((response) => {
-          if (response.data) {
-            setIsFollowing(response.data);
-            setFollowState("following");
-          } else {
-            setFollowState("notFollowing");
-          }
-        })
-        .catch((error) => {
-          console.log(
-            "Failed to get check if current author is following " + error
-          );
+      if (!authorHostIsOurs(author.host) && authorBaseApiURL !== null) {
+        await api
+          .get(`${baseURL}/node/?host=${author.host}`)
+          .then((response) => {
+            let node = createNodeObject(response, author.host);
+            api
+              .post(`${host}authors/${extractAuthorUUID(author.id)}/followers/${currentAuthor.uuid}`, { header: node.headers })
+              .then((response) => {
+                if (response.data) {
+                  setIsFollowing(response.data);
+                  setFollowState("following");
+                } else {
+                  setFollowState("notFollowing");
+                }
+              })
+              .catch((error) => {
+                console.log("Failed to get check if current author is following " + error);
+              });
```

Comment:
Don't need to call the node endpoint, we already have node from the props.
```suggestion
            api
              .post(`${props.authorBaseApiURL}authors/${extractAuthorUUID(author.id)}/followers/${currentAuthor.uuid}`, { header: props.authorNode.headers })
              .then((response) => {
                if (response.data) {
                  setIsFollowing(response.data);
                  setFollowState("following");
                } else {
                  setFollowState("notFollowing");
                }
              })
```